ql/src/test/queries/clientpositive/results_cache_invalidation3.q

```diff
@@ -0,0 +1,16 @@
+set hive.support.concurrency=true;
+set hive.txn.manager=org.apache.hadoop.hive.ql.lockmgr.DbTxnManager;
+
+set hive.query.results.cache.enabled=true;
+set hive.query.results.cache.nontransactional.tables.enabled=false;
```

Comment:
It would be nice to add also another test case (e.g., `results_cache_invalidation4.q`) when `hive.query.results.cache.nontransactional.tables.enabled=true` using non-transactional tables. It seems that if the table is dropped with the new logic we should be able to detect if the cache entry is valid or not.